src/axom/inlet/Table.cpp

```diff
@@ -193,5 +246,95 @@ bool Table::required()
   return (bool)intValue;
 }
 
+ std::shared_ptr<Table> Table::registerVerifier(std::function<bool()> lambda) {
+  SLIC_WARNING_IF(m_verifier, fmt::format("Verifier for Table {0} already set", 
+                                         m_sidreGroup->getPathName()));
+  m_verifier = lambda;
+  return shared_from_this();
+ } 
+
+ bool Table::verify() {
+   bool verified = true;
+   // Verify this Table
+   if (m_verifier && !m_verifier()) {
+     verified = false;
+     SLIC_WARNING(fmt::format("Table {0} failed verification", m_sidreGroup->getPathName()));
+   }
+   // Verify the child Fields of this Table
+   for (auto field : m_fieldChildren) {
+     if (!field.second->verify()) {
+       verified = false;
+     }
+   }
+   // Verify the child Tables of this Table
+   for (auto table : m_tableChildren) {
+     if (!table.second->verify()) {
+       verified = false;
+     }
+   }
+   
+   return verified;
+ }
+
+bool Table::hasField(const std::string& fieldName) {
+  return m_fieldChildren.find(getFullName(m_name, fieldName)) != m_fieldChildren.end();
+}
+
+bool Table::hasTable(const std::string& tableName) {
+  return m_tableChildren.find(getFullName(m_name, tableName)) != m_tableChildren.end();
+}
+
+std::shared_ptr<Table> Table::getTable(std::string name) {
```

Comment:
I feel like getTable and getField should be functions of the Inlet class.

Reply:
Also change the parameter to "const std:string& name" and please don't alter it during the finding.